include/fiber_tasking_lib/task_scheduler.h

```diff
@@ -181,19 +183,21 @@ class TaskScheduler {
 	/**
 	 * Yields execution to another task until counter == value
 	 *
-	 * @param counter    The counter to check
-	 * @param value      The value to wait for
+	 * @param counter             The counter to check
+	 * @param value               The value to wait for
+	 * @param pinToCurrentThread  If true, the task invoking this call will not resume on a different thread
 	 */
-	void WaitForCounter(std::shared_ptr<std::atomic_uint> &counter, uint value);
+	void WaitForCounter(std::shared_ptr<std::atomic_uint> &counter, uint value, bool pinToCurrentThread = false);
 
-private:
 	/**
 	 * Gets the 0-based index of the current thread
 	 * This is useful for m_tls[GetCurrentThreadIndex()]
 	 *
 	 * @return    The index of the current thread
 	 */
 	std::size_t GetCurrentThreadIndex();
+
+private:
```

Comment:
You shouldn't need to promote GetCurrentThreadIndex() to public. The only additional place you use it is in FiberStart, which is a static member function of TaskScheduler. So it can access private functions.